csrc/ops/arith.cpp

```diff
@@ -455,28 +455,57 @@ NVFUSER_DEFINE_UNARY_OP(ceil, Ceil)
 NVFUSER_DEFINE_UNARY_OP(floor, Floor)
 NVFUSER_DEFINE_UNARY_OP(frac, Frac)
 NVFUSER_DEFINE_UNARY_OP(neg, Neg)
-NVFUSER_DEFINE_UNARY_OP(notOp, Not)
 NVFUSER_DEFINE_UNARY_OP(relu, Relu)
 NVFUSER_DEFINE_UNARY_OP(round, Round)
 NVFUSER_DEFINE_UNARY_OP(silu, Silu)
 NVFUSER_DEFINE_UNARY_OP(trunc, Trunc)
 NVFUSER_DEFINE_UNARY_OP(print, Print)
 #undef NVFUSER_DEFINE_UNARY_OP
 
+Val* logical_not(Val* v) {
+  if (!isBooleanType(v->dtype())) {
+    TORCH_CHECK(
+        isIntegralType(v->dtype()),
```

Comment:
I think this don't have to be integral, for example:
```C++
[cling]$ 1.0 && 2.0
input_line_3:2:9: warning: implicit conversion from 'double' to 'bool' changes value from 2 to true [-Wliteral-conversion]
 1.0 && 2.0
     ~~ ^~~
(bool) true
```

Reply:
Good point. I will remove those checks.